rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/MultiSessionAttributeAdapter.java

```diff
@@ -217,7 +219,17 @@ private void invalidateAllSession(HttpSession preferredSession, HttpSession loca
         private HttpSession findValidPreferredSession(HttpSession localSession, HttpServletRequest optionalRequest) {
             if (localSession instanceof Session) {
                 SessionHandler preferredHandler = getPreferredJettyHandler((Session)localSession, true, true);
-                HttpSession preferredSession = preferredHandler==null ? null : preferredHandler.getHttpSession(localSession.getId());
+
+                HttpSession preferredSession
+                if (preferredHandler == null) {
```

Comment:
should be !=null

Reply:
Doh! I'd written it for `== null` then changed the logic and forgot to change the check...